multiraft/transport.go

```diff
@@ -100,6 +103,18 @@ func NewLocalRPCTransport() Transport {
 	}
 }
 
+// EnableEvents creates the Events channel. All future clients will send an
+// *interceptMessage to that channel and block until the message is
+// acknowledged. EnableEvents should be called before any clients are created,
+// i.e. prior to calls to Connect.
+func (lt *localRPCTransport) EnableEvents() {
```

Comment:
I don't think we should build this into localRPCTransport. There should be a separate interceptingTransport type that either wraps an existing Transport or simply is a transport itself (I think the latter is probably better - you don't need the RPC machinery, just have Go call the ServerInterface method directly). Then this can all go in transport_test.go unless and until we have a use for it outside this package.